### PR TITLE
Feature/messege to share

### PR DESCRIPTION
**Pull request by Michael-B-D**
add  a func that give you a nice template to share the event

### CODE REVIEW

File: app/routers/event.py [hunk 1]
```diff
@@ -321,6 +323,25 @@ def add_new_event(values: dict, db: Session) -> Optional[Event]:
...
+def get_tamplate_to_share_event(event_id: int, user_name: str,
+                                db: Session, request: Request) -> templates:
+    """Give yoe nice tamplate of the event that you can share.
+    Get event id and the user name that want to send the share message,
+    Also needed to give db session and request method (mostly 'get')
+    return TemplateResponse that incloud all the ditails on the event"""
+
+    event = by_id(db, event_id)
+    html_temp = templates.TemplateResponse("event/share_event.html",
+                                           {"request": request,
```
Comment (yammesicka) [inline app/routers/event.py:335]: Can you please format it a bit better? It take lots of lines for no good reason :)

File: app/templates/event/partials/edit_event_details_tab.html [hunk 2]
```diff
@@ -13,12 +13,11 @@
 </div>
 
 <div class="form_row">
-  <select name="location_type" required>
-    <option value="" disabled selected>Type</option>
-    <option value="vc_url">VC URL</option>
-    <option value="address">Address</option>
-  </select>
-  <input type="text" name="location" placeholder="VC URL/Location">
+    <input type="text" name="location" placeholder="Location">
```
Comment (yammesicka) [inline app/templates/event/partials/edit_event_details_tab.html:16]: Please add labels

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: app/templates/event/share_event.html [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta1/dist/css/bootstrap.min.css" rel="stylesheet"
+          integrity="sha384-giJF6kkoqNQ00vy+HMDP7azOuL0xtbfIcaT9wjKHr8RbDVddVHyTfAAsrekwKmP1" crossorigin="anonymous">
+</head>
+<body>
+    <div class="share_event card"  style="width: 30rem;">
+            <h5>Hi!</h5>
```
Comment (yammesicka) [inline app/templates/event/share_event.html:13]: Why `h5`?

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: app/templates/event/share_event.html (hunk 3 above)
Comment (Michael-B-D) [reply app/templates/event/share_event.html:13]: that the format from bootstrap, and it looks nice, you don't think?

Comment (yammesicka) [reply app/templates/event/share_event.html:13]: h5 is a title element :) I'm sure you did great work regarding the design, but you should not use `h5` for design purposes. Please use CSS to get the same effect.

File: app/templates/event/share_event.html [hunk 4]
```diff
@@ -0,0 +1,44 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta1/dist/css/bootstrap.min.css" rel="stylesheet"
+          integrity="sha384-giJF6kkoqNQ00vy+HMDP7azOuL0xtbfIcaT9wjKHr8RbDVddVHyTfAAsrekwKmP1" crossorigin="anonymous">
```
Comment (yammesicka) [inline app/templates/event/share_event.html:9]: CSS Should be before `</body>`

File: app/templates/event/share_event.html [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+<style>
+    .msg_title{
+       font-size: 27px
+    }
```
Comment (yammesicka) [inline app/templates/event/share_event.html:13]: Should be in CSS file

File: app/templates/event/share_event.html [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta1/dist/css/bootstrap.min.css" rel="stylesheet"
+          integrity="sha384-giJF6kkoqNQ00vy+HMDP7azOuL0xtbfIcaT9wjKHr8RbDVddVHyTfAAsrekwKmP1" crossorigin="anonymous">
+<style>
+    .msg_title{
+       font-size: 27px
```
Comment (yammesicka) [inline app/templates/event/share_event.html:12]: Use rem

File: app/templates/event/share_event.html [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    }
+</style>
+</head>
+<body>
+    <div class="share_event card"  style="width: 30rem;">
```
Comment (yammesicka) [inline app/templates/event/share_event.html:17]: Don't use inline styles

File: app/routers/event.py [hunk 8]
```diff
@@ -323,6 +323,21 @@ def add_new_event(values: dict, db: Session) -> Optional[Event]:
...
+                                            "sender_name": user_name,
+                                            "event": event})
```
Comment (yammesicka) [inline app/routers/event.py:337]: Can be joined to only one line

File: app/routers/event.py [hunk 9]
```diff
@@ -323,6 +323,21 @@ def add_new_event(values: dict, db: Session) -> Optional[Event]:
         return None
 
 
+def get_tamplate_to_share_event(event_id: int, user_name: str,
```
Comment (yammesicka) [inline app/routers/event.py:326]: template

File: app/routers/event.py [hunk 10]
```diff
@@ -323,6 +323,21 @@ def add_new_event(values: dict, db: Session) -> Optional[Event]:
...
+                                db: Session, request: Request) -> templates:
+    """Give yoe nice tamplate of the event that you can share.
```
Comment (yammesicka) [inline app/routers/event.py:328]:
```python
"""Gives shareable template of the event.

Args:
     event_id: Event to share
     user_name: The user who shares the event
     db: The database to get the event from
     request: The request we got from the user using FastAPI.

Returns:
    Shareable HTML with data from the database about the event.
"""
```

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: app/routers/event.py [hunk 11]
```diff
@@ -380,6 +380,27 @@ def add_new_event(values: dict, db: Session) -> Optional[Event]:
...
+
+Args:
+     event_id: Event to share
+     user_name: The user who shares the event
+     db: The database to get the event from
+     request: The request we got from the user using FastAPI.
+
+Returns:
+    Shareable HTML with data from the database about the event.
+"""
```
Comment (yammesicka) [inline app/routers/event.py:395]: Indent

File: app/routers/event.py [hunk 12]
```diff
@@ -380,6 +380,27 @@ def add_new_event(values: dict, db: Session) -> Optional[Event]:
...
+
+    event = by_id(db, event_id)
+    data = {"sender_name": user_name, "event": event}
```
Comment (yammesicka) [inline app/routers/event.py:398]: Prefer other name, `data` usually considered invalid name

File: app/static/share_event.css [hunk 13]
```diff
@@ -0,0 +1,6 @@
+.msg_title{
+    font-size: 1.5rem;
+ }
```
Comment (yammesicka) [inline app/static/share_event.css:3]: Put a blank line after this line

File: app/templates/event/share_event.html [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta1/dist/css/bootstrap.min.css" rel="stylesheet"
+          integrity="sha384-giJF6kkoqNQ00vy+HMDP7azOuL0xtbfIcaT9wjKHr8RbDVddVHyTfAAsrekwKmP1" crossorigin="anonymous">
+    <link href="/app/static/share_event.css" rel="stylesheet">
+
+</head>
+<body>
+    <div class="share_event card">
+            <p class="msg_title">Hi!</p>
+            <h6 class="card-subtitle mb-2 text-muted">{{ data.sender_name }} want you to know about event he have,</h6>
```
Comment (yammesicka) [inline app/templates/event/share_event.html:16]: Why h6? If it's for the size, write it as a `<p>` element use CSS to format it

File: app/templates/event/share_event.html [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+                {% if data.event.vc_link is not none %}
+                <a href="{{ data.event.vc_link }}" class="card-link">Link to vc here</a>
+                {% endif %}
+                <!-- TODO get link to the event -->
+                <!-- <p>See the event <a href="--" class="card-link">here</a> </p> -->
+            </div>
+            <!-- TODO: get link to register page -->
+            <div class="card-footer text-muted">
+                This app was built by <a href="https://github.com/PythonFreeCourse" class="card-link">"PythonFreeCourse"</a>
+                <a href="linktoregister" class="card-link">Join the app!</a>
```
Comment (yammesicka) [inline app/templates/event/share_event.html:37]: We have such a thing, `url_for` it

**Review by yammesicka — CHANGES_REQUESTED** [review]

**Comment by yammesicka** [comment]
Great start! I've added few insights :)

**Review by yammesicka — APPROVED** [review]

**Pull request merged**